benchmarks/cugraph/standalone/cugraph_bulk_sampling.py

```diff
@@ -0,0 +1,685 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import warnings
+import argparse
+import traceback
+
+from cugraph.testing.mg_utils import (
+    generate_edgelist_rmat,
+    # get_allocation_counts_dask_persist,
+    get_allocation_counts_dask_lazy,
+    sizeof_fmt,
+    get_peak_output_ratio_across_workers,
+    restart_client,
+    start_dask_client,
+    stop_dask_client,
+    enable_spilling,
+)
+
+from cugraph.structure.symmetrize import symmetrize
+from cugraph.experimental.gnn import BulkSampler
+
+import cugraph
+
+from datetime import datetime
+
+import json
+import re
+import os
+import gc
+from time import sleep, perf_counter
+from math import ceil
+
+import pandas as pd
+import numpy as np
+import cupy
+import cudf
+
+import dask_cudf
+import dask.dataframe as ddf
+from dask.distributed import default_client
+
+from typing import Optional, Union, Dict
+
+
+def construct_graph(dask_dataframe):
+    """
+    Args:
+        dask_dataframe:
+            dask_dataframe contains weighted and undirected edges with self
+            loops. Multiple edges will likely be present as well.
+        directed:
+            If True, the graph will be directed.
+        renumber:
+            If True, the graph will be renumbered.
+    Returns:
+        G:  cugraph.Graph
+    """
+    assert dask_dataframe['src'].dtype == 'int64'
+    assert dask_dataframe['dst'].dtype == 'int64'
+    assert dask_dataframe['etp'].dtype == 'int32'
+
+    G = cugraph.MultiGraph(directed=True)
+    G.from_dask_cudf_edgelist(
+        dask_dataframe, source="src", destination="dst", edge_type='etp', renumber=False
+    )
+    return G
```

Comment:
So `cugraph_dgl` requires edge ids for `homogenous` multi graphs and `heterogenous`  multi graphs .   I am curious on how cugraph_pyg is handing this.

Reply:
edge ids aren't required

Reply:
But I can add them if DGL needs them

Reply:
How does PYG differentiate b/w 2 edges between the same src and destination but with different eids ? Or does it just not handle MultiGraphs ?

Reply:
It doesn't need to differentiate unless there are edge properties, all it cares about is the vertex ids.

Reply:
I guess we will have to make adding edge types and edge ids optional, i think if you make adding edge type optional , I can just chime in a follow up PR that addes edge ids  and not block this PR.

Reply:
Same edge can appear multiple times, that is legal

Reply:
But how do you know which edge id it came from ?  Like if i have `edge id-1`  and `edge id-2` with different features , how do you know which edge id those samples came from ?

Reply:
Well, edge type is not optional because that's used to determine vertex type.  It's just the edge ids that are optional.  And also, PyG's GraphStore interface technically doesn't support edge ids or edge properties right now (they're only supported in the native `Data` and `HeteroData` classes).

Reply:
The answer is we don't care on the PyG end which edge it came from

Reply:
papers100M, MAG, etc. don't have edge properties.

Reply:
Ok, I'm going to add the option to output edge type.  Give me a few minutes...

Reply:
> the remote graph (GraphStore) interface

I am not sure where or why we will use ` remote graph `, i always thought that it was a cugraph service thing that we deprecated for GNN use cases . 

Anyways, I guess the idea now is to just update the script by making adding edge types optional. I can do a follow up to add edge ids.

Reply:
By "remote graph", PyG is referring to any backend that is not native, basically.

Reply:
It doesn't literally mean a remote graph in this case.

Reply:
https://pytorch-geometric.readthedocs.io/en/latest/modules/data.html?highlight=remote#remote-backend-interfaces

Reply:
ok, @VibhuJawa, I added adding edge properties as an option that is off by default.  I tested it and it seems to be working correctly.  Please verify that it works with your DGL workflow.

Reply:
Requested some more changes based on more testing